audioutils/fmsynth/test/fmsynth_alsa_test.c

```diff
@@ -0,0 +1,324 @@
+/****************************************************************************
+ * apps/audioutils/fmsynth/test/fmsynth_alsa_test.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdio.h>
+#include <stdint.h>
+#include <termios.h>
+#include <fcntl.h>
+
+#include <alsa/asoundlib.h>
+
+#include <audioutils/fmsynth_eg.h>
+#include <audioutils/fmsynth_op.h>
+#include <audioutils/fmsynth.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define FS (48000)
+
+#define CHANNEL_NUM (2)
+#define RESAMPLING_ALSA (1)
+#define LATENCY_ALSA (10000)
+
+#define SAMPLE_NUM (FS / 20)
+#define BUFF_LENGTH (SAMPLE_NUM * CHANNEL_NUM)
+
+#define CODE_C_FREQ (261.625565f)
+#define CODE_D_FREQ (293.6647674f)
+#define CODE_E_FREQ (329.6275561f)
+#define CODE_F_FREQ (349.2282305f)
+#define CODE_G_FREQ (391.9954347f)
+#define CODE_A_FREQ (440.f)
+#define CODE_B_FREQ (493.8833009f)
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static int16_t samples[BUFF_LENGTH];
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * name: init_alsa
+ ****************************************************************************/
+
+static snd_pcm_t *init_alsa(int fs)
+{
+  int ret;
+  snd_pcm_t *hndl = NULL;
```

Comment:
FAR

Reply:
This code is just for unit testing on linux.
So I don't think it needs "FAR".